engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -54,10 +55,18 @@ public BlockRegion(AABBi source) {
         aabb.set(source);
     }
 
+    /**
+     * Deprecated in favor of {@link org.terasology.world.block.BlockRegions#createFromMinAndMax(Vector3ic, Vector3ic)}
+     */
+    @Deprecated
```

Comment:
(_copying from Discord for the record_)

I think it is somewhat confusing to have some variants as constructor `new BlockRegion(min, max)` and some as constructor method `BlockRegions.createFromCenterAndExtents(center, extents)`. Especially if another variant like `BlockRegions.createFromMinAndSize(min, size)` has the same argument types...

Most of the implementations are pretty neat one-liners which may be inlined instead. For consistency I'd also inline the two constructor variants that were deprecated.

Only culprit we found is create a BlockRegion from center and extents when given floating point coordinates. One needs to be careful with rounding in that case...

```java
    public static BlockRegion createFromCenterAndExtents(Vector3fc center, Vector3fc extents) {
        Vector3f min = center.sub(extents, new Vector3f());
        Vector3f max = center.add(extents, new Vector3f());

        return new BlockRegion().setMin(new Vector3i(min, RoundingMode.CEILING)).setMax(new Vector3i(max,
                RoundingMode.FLOOR));
    }
```

Reply:
Its a little confusing to move the actual declaration to a separate static class. might be better to move the static implementations into BlockRegion so they are easily found in one place and leave the default constructor for min and max since that directly describes how a region is defined.

Reply:
@pollend From what I read in #4200 you agreed to what we implemented :thinking: 
Why the change of heart?
On what grounds should creation from min and max be the default constructor?

Reply:
I was thinking about it a bit, but maybe it's not as confusing as I thought it would be.

Reply:
One more thing is discoverability from user's perspective. Following up to @jdrueckert's question on why initialization from min and max is the default, I'd argue that from center and extents or from min and size are equally valid ways to create a BlockRegion. 

The pattern with the plural form `BlockRegions` is also used in other places in the Java world (like `Lists` or `Streams` from guava), but there are also cases like `Stream.of` from plain Java... 

I would go for the following line-up (I'd prefer `BlockRegions` but I could also live with `BlockRegion`):

```java
BlockRegion r = new BlockRegion(); // empty region
BlockRegion r = otherRegion.copy()

BlockRegion r = BlockRegions.createFromMinAndMax(min, max)
BlockRegion r = BlockRegions.createFromCenterAndExtents(center, extents)
BlockRegion r = BlockRegions.createFromMinAndSize(min, size)
BlockRegion r = BlockRegions.createFromPoint(point)
```